src/utils/retry.test.ts

```diff
@@ -0,0 +1,33 @@
+import { retry } from './retry'
+
+describe('retry function', () => {
+  jest.useRealTimers()
+  let mockFn: jest.Mock
```

Comment:
This isn't shared state. You should re-declare it in each test: `const mockFn = jest.fn().mockResolvedValue(...)`. Trying to share it makes it harder to read the tests in isolation.